src/script_opt/ZAM/Driver.cc

```diff
@@ -404,16 +404,19 @@ void ZAMCompiler::Dump() {
         }
     }
     else if ( analysis_options.dump_final_ZAM ) {
-        ASSERT(remapped_frame);
-
         printf("\nFrame for %s:\n", func->GetName().c_str());
 
-        for ( auto i = 0U; i < shared_frame_denizens.size(); ++i ) {
-            printf("frame[%d] =", i);
-            for ( auto& id : shared_frame_denizens[i].ids )
-                printf(" %s", id->Name());
-            printf("\n");
+        if ( remapped_frame ) {
```

Comment:
I wonder if adding `"remapped_frame=%d", remapped_frame` to above printf would add a bit more information to the output when remapping happened and when it didn't.

(...without actually knowing what that means :-) )

Reply:
The frame is always remapped unless low-level ZAM optimization is turned off (which is what `-O no-ZAM-opt` does). As this is output for the maintainer, it should be immediately apparent to them whether the frame has been remapped.

BTW, remapping refers to where local variables reside in the frame. Initially each is mapped to its own slot (like the interpreter does). ZAM's low-level optimization identifies which of these don't overlap and thus can share a slot. This often makes the frame much smaller due to the heavy use in script optimization of temporary variables, which turns out to have appreciable performance gains.